gunrock/graph/build.hxx

```diff
@@ -113,29 +117,32 @@ auto from_csr_t(typename vertex_vector_t::value_type const& r,
   } else {
     host::csr_t<graph_type>(G, memory::raw_pointer_cast(O.data()));
   }
+  
+  // Meta
+  constexpr memory_space_t h_space = memory_space_t::host;
 
-  return O;
-}
-
-template <typename vertex_t, typename edge_t>
-auto meta_graph(vertex_t const& r, vertex_t const& c, edge_t const& nnz) {
-  using vertex_type = vertex_t;
-  using edge_type = edge_t;
-  using weight_type = edge_t;
+  using meta_type = graph::graph_t<
```

Comment:
We don't need a whole graph here for sure. Let's think of a simple struct for meta data.
```
struct graph_meta_t {
num_vertices;
num_edges;
...
};

Reply:
Implemented in https://github.com/bkj/essentials/pull/5 -- but after implementing I actually think leaving it as a whole graph makes more sense, to make sure that the interface to `meta` on CPU and `G` on GPU are identical.